dashboard/src/components/api/che-workspace.factory.js

```diff
@@ -317,10 +325,62 @@ export class CheWorkspace {
     }
 
     return 'ws://' + wsagentServerAddress + '/' + contextPath + '/ext/ws/' + workspaceId;
-
   }
 
   getIdeUrl(workspaceName) {
     return '/ide/' + workspaceName;
   }
+
```

Comment:
please add jsdoc for these methods of API : startUpdateWorkspaceStatus, addStatusListener and removeStatusListener


Also, the listener couldn't be setup automatically ?